icu4c/source/i18n/collationbuilder.cpp

```diff
@@ -1182,10 +1188,12 @@ CollationBuilder::addTailComposites(const UnicodeString &nfdPrefix, const Unicod
                                      newNFDString, newString, errorCode)) {
             continue;
         }
-        int32_t newCEsLength = dataBuilder->getCEs(nfdPrefix, newNFDString, newCEs, 0);
+        int32_t newCEsLength = dataBuilder->getCEs(nfdPrefix, newNFDString, newCEs, 0, errorCode);
         if(newCEsLength > Collation::MAX_EXPANSION_LENGTH) {
             // Ignore mappings that we cannot store.
             continue;
+        } else if (U_FAILURE(errorCode)) {
```

Comment:
no need to put the if after "else" since the previous block will always continue anyway.